File: problems/problems_1912/Solution.cpp
//go:build ignore
#include "cpp/common/Solution.h"


using namespace std;
using json = nlohmann::json;

class MovieRentingSystem {
public:
    MovieRentingSystem(int n, vector<vector<int>>& entries) {
        
    }
    
    vector<int> search(int movie) {
        
    }
    
    void rent(int shop, int movie) {
        
    }
    
    void drop(int shop, int movie) {
        
    }
    
    vector<vector<int>> report() {
        
    }
};

/**
 * Your MovieRentingSystem object will be instantiated and called as such:
 * MovieRentingSystem* obj = new MovieRentingSystem(n, entries);
 * vector<int> param_1 = obj->search(movie);
 * obj->rent(shop,movie);
 * obj->drop(shop,movie);
 * vector<vector<int>> param_4 = obj->report();
 */

json leetcode::qubh::Solve(string input_json_values) {
	vector<string> inputArray;
	size_t pos = input_json_values.find('\n');
	while (pos != string::npos) {
		inputArray.push_back(input_json_values.substr(0, pos));
		input_json_values = input_json_values.substr(pos + 1);
		pos = input_json_values.find('\n');
	}
	inputArray.push_back(input_json_values);

	vector<string> operators = json::parse(inputArray[0]);
	vector<vector<json>> op_values = json::parse(inputArray[1]);
	auto obj0 = make_unique<MovieRentingSystem>(op_values[0][0], op_values[0][1]);
	vector<json> ans = {nullptr};
	for (size_t i = 1; i < op_values.size(); ++i) {
		if (operators[i] == "search") {
			ans.push_back(obj0->search(op_values[i][0]));
			continue;
		}
		if (operators[i] == "rent") {
			obj0->rent(op_values[i][0], op_values[i][1]);
			ans.push_back(nullptr);
			continue;
		}
		if (operators[i] == "drop") {
			obj0->drop(op_values[i][0], op_values[i][1]);
			ans.push_back(nullptr);
			continue;
		}
		if (operators[i] == "report") {
			ans.push_back(obj0->report());
			continue;
		}
		ans.push_back(nullptr);
	}
	return ans;
}

File: problems/problems_1912/Solution.java
package problems.problems_1912;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


class MovieRentingSystem {

    public MovieRentingSystem(int n, int[][] entries) {
        
    }
    
    public List<Integer> search(int movie) {
        
    }
    
    public void rent(int shop, int movie) {
        
    }
    
    public void drop(int shop, int movie) {
        
    }
    
    public List<List<Integer>> report() {
        
    }
}

/**
 * Your MovieRentingSystem object will be instantiated and called as such:
 * MovieRentingSystem obj = new MovieRentingSystem(n, entries);
 * List<Integer> param_1 = obj.search(movie);
 * obj.rent(shop,movie);
 * obj.drop(shop,movie);
 * List<List<Integer>> param_4 = obj.report();
 */

public class Solution extends BaseSolution {


    @Override
    public Object solve(String[] inputJsonValues) {
        String[] operators = jsonArrayToStringArray(inputJsonValues[0]);
		String[][] opValues = jsonArrayToString2DArray(inputJsonValues[1]);
		int n = Integer.parseInt(opValues[0][0]);
		int[][] entries = jsonArrayToInt2DArray(opValues[0][1]);
		MovieRentingSystem obj = new MovieRentingSystem(n, entries);
		List<Object> ans = new ArrayList<>(operators.length);
		ans.add(null);
		for (int i = 1; i < operators.length; i++) {
			if (operators[i].compareTo("search") == 0) {
				int movie = Integer.parseInt(opValues[i][0]);
				ans.add(obj.search(movie));
				continue;
			}
			if (operators[i].compareTo("rent") == 0) {
				int shop = Integer.parseInt(opValues[i][0]);
				int movie = Integer.parseInt(opValues[i][1]);
				obj.rent(shop, movie);
				ans.add(null);
				continue;
			}
			if (operators[i].compareTo("drop") == 0) {
				int shop = Integer.parseInt(opValues[i][0]);
				int movie = Integer.parseInt(opValues[i][1]);
				obj.drop(shop, movie);
				ans.add(null);
				continue;
			}
			if (operators[i].compareTo("report") == 0) {
				
				ans.add(obj.report());
				continue;
			}
			ans.add(null);
		}
        return JSON.toJSON(ans);
    }
}
